File: src/main/java/gargoyle/ct/convert/impl/BooleanConverter.java
package gargoyle.ct.convert.impl;

import gargoyle.ct.convert.Converter;
import org.jetbrains.annotations.NotNull;

public class BooleanConverter implements Converter<Boolean> {
    @Override
    public String format(Boolean data) {
        return String.valueOf(data);
    }

    @NotNull
    @Override
    public Boolean parse(String data) {
        return Boolean.valueOf(data);
    }
}

File: src/main/kotlin/gargoyle/ct/convert/impl/BooleanConverter.kt
package gargoyle.ct.convert.impl

import gargoyle.ct.convert.Converter

class BooleanConverter : Converter<Boolean> {
    override fun format(data: Boolean): String {
        return data.toString()
    }

    override fun parse(data: String): Boolean {
        return data.toBoolean()
    }
}
